src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -395,13 +395,22 @@ ep_rt_aot_current_process_get_id (void)
     return static_cast<uint32_t>(GetCurrentProcessId ());
 }
 
+#ifdef TARGET_UNIX
+inline uint64_t THREADSilentGetCurrentThreadId()
+{
+    static __thread uint64_t tid;
+    if (!tid)
+        tid = PalGetCurrentOSThreadId();
```

Comment:
Wouldn't an InterlockedIncrement over a static variable be cheaper here?

Reply:
We need the actual OS thread ID. The threadID is written into the trace. It cannot be just some random number.